src/test/kotlin/uk/gov/dluhc/printapi/client/ElectoralRegistrationOfficeManagementApiClientTest.kt

```diff
@@ -53,144 +57,218 @@ internal class ElectoralRegistrationOfficeManagementApiClientTest {
+            val eroId = aValidRandomEroId()
+
+            val eroResponse = buildElectoralRegistrationOfficeResponse()
+            given(clientResponse.bodyToMono(ElectoralRegistrationOfficeResponse::class.java)).willReturn(
+                Mono.just(eroResponse)
             )
-        )
-
-        val erosResponse = ElectoralRegistrationOfficesResponse(listOf(eroResponse))
-        given(clientResponse.bodyToMono(ElectoralRegistrationOfficesResponse::class.java)).willReturn(
-            Mono.just(erosResponse)
-        )
-        val expected = with(eroResponse) {
-            EroManagementApiEroDto(
-                id = id,
-                name = name,
-                listOf(
-                    EroManagementApiLocalAuthorityDto(
-                        gssCode = localAuthorities[0].gssCode!!,
-                        name = localAuthorities[0].name!!
-                    ),
-                    EroManagementApiLocalAuthorityDto(
-                        gssCode = localAuthorities[1].gssCode!!,
-                        name = localAuthorities[1].name!!
-                    ),
-                ),
-                englishContactDetails = anEnglishEroContactDetails(),
-                welshContactDetails = aWelshEroContactDetails(),
+            val expected = buildEroManagementApiEroDto()
+            given(eroMapper.toEroManagementApiEroDto(any())).willReturn(expected)
+
+            // When
+            val ero = apiClient.getElectoralRegistrationOffice(eroId)
+
+            // Then
+            verify(eroMapper).toEroManagementApiEroDto(eroResponse)
+            assertThat(ero).isEqualTo(expected)
+            assertThat(clientRequest.value.url()).hasHost("ero-management-api").hasPath("/eros/$eroId")
+        }
+
+        @Test
+        fun `should not get Electoral Registration Office given API returns a 404 error`() {
+            // Given
+            val eroId = aValidRandomEroId()
+
+            val http404Error = NOT_FOUND.toWebClientResponseException()
+            given(clientResponse.bodyToMono(ElectoralRegistrationOfficeResponse::class.java)).willReturn(
+                Mono.error(http404Error)
+            )
+
+            val expectedException = ElectoralRegistrationOfficeNotFoundException(mapOf("eroId" to eroId))
+
+            // When
+            val ex = catchThrowableOfType(
+                { apiClient.getElectoralRegistrationOffice(eroId) },
+                ElectoralRegistrationOfficeNotFoundException::class.java
             )
+
+            // Then
+            assertThat(ex.message).isEqualTo(expectedException.message)
+            assertThat(clientRequest.value.url()).hasHost("ero-management-api").hasPath("/eros/$eroId")
         }
-        given(eroMapper.toEroManagementApiEroDto(any())).willReturn(expected)
 
-        // When
-        val ero = apiClient.getElectoralRegistrationOffice(gssCode)
+        @Test
+        fun `should not get Electoral Registration Office given API returns a 500 error`() {
```

Comment:
@Neil-Massey , do you think it makes sense to cover the non-WebClientResponseException case as well, similar to this test `should throw exception given API returns a non WebClientResponseException` below?

Reply:
Yes, this is missing from source reference: https://github.com/cabinetoffice/eip-ero-voter-card-applications-api/blob/main/src/test/kotlin/uk/gov/dluhc/votercardapplicationsapi/client/ElectoralRegistrationOfficeManagementApiClientTest.kt